pdfly/cli.py

```diff
@@ -37,6 +38,29 @@ def version_callback(value: bool) -> None:
     ),
     rich_markup_mode="rich",  # Allows to pretty-print commands documentation
 )
+@entry_point.command(name="sign", help="Sign a PDF file using a PKCS12 (.p12) certificate.")
+def sign(
+    input_pdf: Path = typer.Argument(..., help="Input PDF file."),
```

Comment:
```suggestion
    input_pdf: Annotated[
        Path,
        typer.Argument(
            exists=True,
            file_okay=True,
            readable=True,
            resolve_path=True,
            help="Input PDF file.",
        ),
    ],
```
As can be seen in other subcommands, `typer` provides those `Annoted` types than defines some basic checks on the input file(s).
They are handy, and are a good substitute to the "manual" check that you performed below: `if not p12 or not p12.exists(): ...`

Reply:
it will show:"AttributeError: 'NoneType' object has no attribute 'isidentifier'"

Reply:
Sorry but I do not understand this comment 😅

Maybe you wanted to reply in the thread below regarding `fitz` & `fpdf2`?
If so, just providing the error is not enough for me to figure it out.
You will have to share the code you used.